tests/testthat/test-problems.R

```diff
@@ -150,13 +150,15 @@ test_that("problems return the proper row number", {
 test_that("can promote vroom parse warning to error", {
   make_warning <- function() {
     x <- vroom(
-      I("a,b\n1.0,x\n"),
+      I("a\nx\n"),
       delim = ",",
-      col_types = "dd"
+      col_types = "d",
+      altrep = TRUE
     )
 
-    # Warning happens at print time so force a print
-    print(x)
+    # do something to trigger the warning using R's internal C code
+    # https://github.com/r-lib/cpp11/issues/274
+    x$a[1]
```

Comment:
After rereading my in depth issue, how do you feel about this block of comments?

I think you could remove all the other comments in and around this test-that block, and put this block of comments right after the `test_that()` line and before the `make_warning <- ` line

```suggestion
    # Trigger vroom parse warning while inside R's internal C code for `[` and ensure it doesn't crash R.
    # `[` -> R's C function `do_subset()` -> ALTREP calls `vroom::real_Elt()` -> `vroom::warn_for_errors()`
    # `warn_for_errors()` warns with base R's machinery rather than cpp11's to avoid calling
    # `unwind_protect()` (which throws on longjmp, i.e. on `abort()`) while inside R's internal
    # C code (which doesn't catch C++ exceptions).
    # https://github.com/r-lib/cpp11/issues/274
    # https://github.com/tidyverse/vroom/pull/441#discussion_r883611090
    x$a[1]
```